### PR TITLE
Feat/stylegan imfmodel (wip)

### PR DESCRIPTION
**Pull request by johndpope**
params resnet requires grad = false
fix for incorrect model 
clean up imfmodel - allow inference
add ema to training loop

https://wandb.ai/snoozie/IMF/runs/rr4lmfpn?nw=nwusersnoozie

### CODE REVIEW

**Comment by johndpope** [comment]
@tanshuai0219  - i make some CRITICAL updates to dimensions. 
There's no mention 128 - the diagram has up arrow - so its 256. 
all the previous code (WRONG) was using rectangle for attention - training starts off ok - but then falls apart. I'm hoping these updates resolve things. 
https://wandb.ai/snoozie/IMF/runs/4k2q63b2?nw=nwusersnoozie

UPDATE - looks like it's finally working
https://wandb.ai/snoozie/IMF/runs/zh1o9mo0?nw=nwusersnoozie

im not sure if the resnetfeatureextractor is better than densefeatureencoder - jury is out. 
did you use nni / microsoft? i was thinking of plugging in different models and having feature space solve the best model. 
also - I saw your emotion ed-talk - looks great.
did you see this paper called IRFD? https://arxiv.org/pdf/2405.07257
https://github.com/johndpope/speak-hack

@varunjain99 - check this out 
https://openaccess.thecvf.com/content/CVPR2024/papers/Gao_Implicit_Motion_Function_CVPR_2024_paper.pdf
 

UPDATE
something not right - i didn't the commit changes or the git commit thats stored doesn't recreate the loss - - need to craft a script that guarantees the code is correct - when the feature / motion dimensions all line up - then the loss is tiny - https://wandb.ai/snoozie/IMF/runs/zh1o9mo0/logs
